### DISCUSSION TITLE
Projects (beta): Make it easier to add issues / convert existing projects

### DISCUSSION THREAD

**Discussion by davestewart** [community/community#5493] [Projects and Issues] [answered]
Great to finally try the new Projects Beta.

I'm not sure how valuable this will be, but here are my thoughts around adding items to projects.

I hope it doesn't come across at all complain-y; it's just meant to give an idea of one developer's workflow 👍

## Adding projects

I see that the UX to add issues to projects is a search field, where you have to:

1. click +
2. select a repo
3. type #
4. wait for filtered list to load
5. select an issue
6. repeat (from step 3)

Once you get into the repeating bit, it's reasonably quick, but:

- there's not much context in that little filter list
- if you don't know what you want to add, there's very little context
- some issues don't seem to show

Unless I've missed something – and I know this is a Beta – this feels like an overly-advanced and context-lacking way to add issues to a project.

It's just not the way I think about "what to do next".

Rather I would:

- browse the main issues list, and when I see something I like, click in and assign it a project
- filter the issues list by keyword, the repeat the above
- browse the issues list and assign a label or milestone, then use the `<type>:<name>` filter in Project: Add Cards

The ideal (and simple!) way to add issues to projects would be to use the checkboxes in the issues list, and just assign.

However, strangely this functionality has always been missing (though you can assign to labels, milestones and people) !. 

Anyway; point being, without having the proper context of the list as a whole, I really can't see myself adding items to projects using the filter mechanism (though it is useful when typing in issues; it just feels very limited in the fully-fledged project UI).

## Converting projects

Additionally, most of my projects are already set up; I have ~200 issues right now, all of which have been assigned a milestone or project, and I don't really relish the idea of adding them one-by-one to a new project.

## Areas / Projects vs Milestones

At the moment I bastardize Milestones as a poor-man's Areas.

I was hoping that "Areas" in the new Projects would be a way for me to assign things which fall outside labels, and between projects and milestones, but until there's...

- a way to quickly convert projects, or add items more easily
- a way to assign Areas using Milestones (unlikely!)

...the overhead of converting all my existing issues and projects probably doesn't warrant the effort.

**Comment by chrisreddington** [comment]
> Additionally, most of my projects are already set up; I have ~200 issues right now, all of which have been assigned a milestone or project, and I don't really relish the idea of adding them one-by-one to a new project.

I only had about 100 issues, but I went through and added these one-by-one (it felt a little long-winded). Would be great to have a method of bulk importing existing issues to a project.

**Comment by davestewart** [comment]
For the folks actually designing the new Projects, the other weird thing about Projects is actually you **don't** really want every issue in the so-called "backlog" to be ***in*** a project.

I would much rather be able to categorise my Issues using Areas (as I said, I currently use Milestones for this purpose) but add them to a project _**only**_ when I want to work on them.

Right now (with old Projects) the only reason I add them to a project is so they don't get forgotten (and because I keep flip-flopping between Milestones and Projects because neither are the right fit).

**I hope we will also be able to categorise Issues with Areas** which should allow us to leave them in the Issues list until I am ready to pick them for a Project:

![the claw](https://media.giphy.com/media/ciadMxfm3135m/giphy.gif?cid=ecf05e4757iizm1f0qpsx5c6km2s24voc13l11iymnfs8d6z&rid=giphy.gif&ct=g)

**Comment by elyobo** [comment]
Related to this: the repo filter is persisted when adding an issue, but no other are, e.g. when adding a bunch of issues to a new repo the `is:open` is useful for dropping down the noise level (I assume wanting to add merged PRs and closed issues is relatively uncommon) and yet it's dropped off after I choose an issue.

**Reply by elyobo** [reply]
Urgh, and `is:open` no longer even seems to work as a filter.

**Comment by xnacly** [comment]
I agree we at @fosscord would love to use this feature to improve our current workflows to assign maintainers, edit issues and to get an overview over issues.
Currently we are doing the same thing by getting issues from all repos across the orga and inserting them into a notion board.
The problem with this approach is the missing ability for a full sync. This means we cant edit tags, title or assingees from the notion overview. 
Our approach is working but we would love to use the github projects beta to be more productive.

**Comment by mattfitzgerald** [comment]
Hi @davestewart 

Note that instead of this...

> I see that the UX to add issues to projects is a search field, where you have to:
> 
> 1. click +
> 2. select a repo
> 3. type #
> 4. wait for filtered list to load
> 5. select an issue
> 6. repeat (from step 3)

you can add an issue to a project on the issue itself.  (Click a project to toggle inclusion.)

![Screen Shot 2021-10-11 at 12 43 09 pm](https://user-images.githubusercontent.com/955923/136721842-dcc5255c-02c5-48ba-858d-fc5f41534ebf.png)

Not quite as slick as selecting many from a list and adding all at once but probably better than adding from within the project view as you describe.

**Reply by davestewart** [reply]
Ah, the new project seems to be hidden away under the "Organisation" tab: 

![image](https://user-images.githubusercontent.com/132681/136789506-205c658c-aa18-4411-bac0-abe983de18a4.png)

I do this generally for issues when I create them, but as I mentioned in the original post, I don't see why GH don't just implement checkboxes and bulk assign; waaaaay easier and quicker than these newfangled UIs they seem to want to come up with.

**Reply by davestewart** [reply]
BTW, do you know if the new projects are _only_ organisation only? Can they be repo only?

**Comment by mobile-sergey** [comment]
Some of needs are in roadmap:

- [Automatically add new issues from a repository into projects #286](https://github.com/github/roadmap/issues/286)
- [Projects beta parity with projects classic #287](https://github.com/github/roadmap/issues/287)
- [Ability to add a project to a project #274](https://github.com/github/roadmap/issues/274)

But all this features are added to Future, after Q3 2022 – Jul-Sep: [GitHub public roadmap](https://github.com/github/roadmap/projects/1)

**Reply by davestewart** [reply]
Thanks for that.

Still waiting on functionality to check multiple issues and bulk add them to a project, sadly.

**Comment by omerbensaadon** [comment, marked as answer]
Hey, I think the features you're looking for has been released! 

We call this Bulk Add, and you can do it from the Repo Issue List as you're describing 💯 

See change-log: [github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update](https://github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update/)

**Reply by mobile-sergey** [reply]
Yes, I saw it on this week. Thanks.

**Reply by davestewart** [reply]
This is such great news!!!

Thanks to you and the team for listening to the feedback and acting upon it. I think it's certainly a "missing" feature rather than some newfangled one, so congrats and we can all sleep soundly in our beds now that is ticked off ✅    🎉   😋   🙏

**Comment by davestewart** [comment]
Hey!

Just checking back in on this.

So it turns out the projects dropdown for checked issues works differently to how it does for labels, milestones, users, etc.

**EDIT**: this might be a bug, as the GIF in the [blog post](https://github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update/) certainly shows it as working the way it was discussed.

For labels etc, you see all available entities, with checked / indeterminate states for checked issues, and can then assign or unassign the checked items by selecting the items in the list:

![image](https://user-images.githubusercontent.com/132681/166566671-0f4366aa-10a7-4fbf-9da6-ac22a881454b.png)

For projects, it's a filter only; having conflicting items checked means the list shows empty:

![image](https://user-images.githubusercontent.com/132681/166566629-31a3861f-4253-4904-ab48-9b7e42da3874.png)

You need to uncheck any checked issues in order to see all projects:

![image](https://user-images.githubusercontent.com/132681/166566812-152acca5-e933-42f4-9e10-a87013971552.png)

Then then choose the project to filter the list:

![image](https://user-images.githubusercontent.com/132681/166566944-11bd532a-77dc-400d-8e13-cd88ec2f1d68.png)

The upshot is that you can't use this method to bulk-assign issues to a project as the original discussion was requesting.

Not sure if this will get any eyes on it now, so just @-ing you @omerbensaadon.

**Comment by davestewart** [comment]
@omerbensaadon - also, I just realised the "checkbox > assign to project" functionality is not implemented in Milestones, which is a real shame as I (ab)use Milestones as pseudo-backlogs for Projects.

The reason I do this is because it's easier to categorise an issue when it is created, and when I create a project, I don't just want to fill up the Todo column, as I want the Project as a whole to be a manageable amount of work.
